spec/requests/api/provision_requests_spec.rb

```diff
@@ -95,6 +95,155 @@
+    end
+    let(:flavor) do
+      FactoryGirl.create(:flavor_amazon, :ems_id => ems.id, :name => 't2.small', :cloud_subnet_required => true)
+    end
+    let(:az)             { FactoryGirl.create(:availability_zone_amazon, :ems_id => ems.id) }
+    let(:cloud_network1) { FactoryGirl.create(:cloud_network_amazon, :ems_id => ems.network_manager.id, :enabled => true) }
+    let(:cloud_subnet1) do
+      FactoryGirl.create(:cloud_subnet, :ems_id => ems.id, :cloud_network => cloud_network1, :availability_zone => az)
+    end
+    let(:security_group1) do
+      FactoryGirl.create(:security_group_amazon, :name => "sgn_1", :ext_management_system => ems,
+                         :cloud_network => cloud_network1)
+    end
+    let(:floating_ip1) do
+      FactoryGirl.create(:floating_ip_amazon, :cloud_network_only => true, :ems_id => ems.network_manager.id,
+                         :cloud_network => cloud_network1)
+    end
+
+    let(:provreq_body) do
+      {
+        "template_fields" => {"guid" => template.guid},
+        "requester"       => {
+          "owner_first_name" => "John",
+          "owner_last_name"  => "Doe",
+          "owner_email"      => "user@example.com"
+        }
+      }
+    end
+
+    let(:expected_provreq_attributes) { %w(id options) }
+
+    let(:expected_provreq_hash) do
+      {
+        "userid"         => api_config(:user),
+        "requester_name" => api_config(:user_name),
+        "approval_state" => "pending_approval",
+        "type"           => "MiqProvisionRequest",
+        "request_type"   => "template",
+        "message"        => /Provisioning/i,
+        "status"         => "Ok"
+      }
+    end
+
+    it "supports manual placement" do
+      api_basic_authorize collection_action_identifier(:provision_requests, :create)
+
+      body = provreq_body.merge(
+        "vm_fields" => {
+          "vm_name"                     => "api_test_aws",
+          "instance_type"               => flavor.id,
+          "placement_auto"              => false,
+          "placement_availability_zone" => az.id,
+          "cloud_network"               => cloud_network1.id,
+          "cloud_subnet"                => cloud_subnet1.id,
+          "security_groups"             => security_group1.id,
+          "floating_ip_address"         => floating_ip1.id
+        }
+      )
+
+      run_post(provision_requests_url, body)
+
+      expect_request_success
+      expect_result_resources_to_include_keys("results", expected_provreq_attributes)
+      expect_results_to_match_hash("results", [expected_provreq_hash])
+
+      expect(response_hash["results"].first).to a_hash_including(
+        "options" => a_hash_including(
+          "placement_auto"              => [false, 0],
+          "placement_availability_zone" => [az.id, az.name],
+          "cloud_network"               => [cloud_network1.id, cloud_network1.name],
+          "cloud_subnet"                => [cloud_subnet1.id, anything],
```

Comment:
had to use `anything` here because I can't find how subnets are serialized here, the format is : `"cloud_subnet_0000000000001 () | availability_zone_0000000000001"`